client/debug.go

```diff
@@ -0,0 +1,43 @@
+package client
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+)
+
+type autoCloseResponses struct {
+	transport http.RoundTripper
+}
+
+func (a *autoCloseResponses) RoundTrip(req *http.Request) (*http.Response, error) {
+	resp, err := a.transport.RoundTrip(req)
+	resp.Body = &autoClosingBody{delegate: resp.Body, timer: time.AfterFunc(time.Minute, func() { autoCloseReport(req, resp) })}
+	return resp, err
+}
+
+type autoClosingBody struct {
+	delegate io.ReadCloser
+	timer    *time.Timer
+}
+
+func (a *autoClosingBody) Read(buf []byte) (int, error) {
+	return a.delegate.Read(buf)
```

Comment:
Ah, I had done that at one point but things weren't working (obviously due to some other reason, but I was ruling stuff out). I'll patch that up once I'm finished with my current rabbit hole.